### PR TITLE
[13.0][FIX] account_invoice_report_grouped_by_picking: refund invoice report

### PR DESCRIPTION
**Pull request by chienandalu**
Consider pending pickings that are linked to the invoice.

cc @Tecnativa TT40412

please review @sergio-teruel @carlosdauden

### CODE REVIEW

File: account_invoice_report_grouped_by_picking/models/account_move.py [hunk 1]
```diff
@@ -15,7 +15,12 @@ class AccountMove(models.Model):
     @api.model
     def _sort_grouped_lines(self, lines_dic):
         return sorted(
-            lines_dic, key=lambda x: (x["picking"].date, x["picking"].date_done)
+            lines_dic,
+            key=lambda x: (
+                x["picking"].date_done
+                and (x["picking"].date, x["picking"].date_done)
+                or (x["picking"].date, x["picking"].date_done)
```
Comment (pedrobaeza) [inline account_invoice_report_grouped_by_picking/models/account_move.py]: Isn't this the same as the previous line?

Comment (chienandalu) [reply account_invoice_report_grouped_by_picking/models/account_move.py]: Sure, it wasn't the intention. Thanks :S

Comment (chienandalu) [reply account_invoice_report_grouped_by_picking/models/account_move.py]: Fixed!

**Review by pedrobaeza — APPROVED** [review]

File: account_invoice_report_grouped_by_picking/models/account_move.py [hunk 2]
```diff
@@ -15,7 +15,12 @@ class AccountMove(models.Model):
     @api.model
     def _sort_grouped_lines(self, lines_dic):
         return sorted(
-            lines_dic, key=lambda x: (x["picking"].date, x["picking"].date_done)
+            lines_dic,
+            key=lambda x: (
+                x["picking"].date_done
+                and (x["picking"].date, x["picking"].date_done)
+                or (x["picking"].date, x["picking"].date)
+            ),
```
Comment (carlosdauden) [inline account_invoice_report_grouped_by_picking/models/account_move.py]:
```suggestion
            lines_dic, key=lambda x: (x["picking"].date, x["picking"].date_done or x["picking"].date)
```

File: account_invoice_report_grouped_by_picking/models/account_move.py [hunk 3]
```diff
@@ -15,7 +15,12 @@ class AccountMove(models.Model):
     @api.model
     def _sort_grouped_lines(self, lines_dic):
         return sorted(
-            lines_dic, key=lambda x: (x["picking"].date, x["picking"].date_done)
+            lines_dic,
+            key=lambda x: (
+                x["picking"].date_done
+                and (x["picking"].date, x["picking"].date_done)
+                or (x["picking"].date, x["picking"].date)
```
Comment (carlosdauden) [inline account_invoice_report_grouped_by_picking/models/account_move.py]: Another option:
```suggestion
                or x["picking"].date
```

**Review by carlosdauden — CHANGES_REQUESTED** [review]

File: account_invoice_report_grouped_by_picking/models/account_move.py (hunk 3 above)
Comment (chienandalu) [reply account_invoice_report_grouped_by_picking/models/account_move.py]: This one wouldn't be valid as we would have a mix of tuples and datetimes.

File: account_invoice_report_grouped_by_picking/models/account_move.py (hunk 2 above)
Comment (chienandalu) [reply account_invoice_report_grouped_by_picking/models/account_move.py]: I went with this suggestion

**Review by carlosdauden — APPROVED** [review]

**Pull request merged**